### DISCUSSION TITLE
Jotai suspends unnecessarily when a fullfilled promise is used by a derived atom

### DISCUSSION THREAD

**Discussion by scamden** [pmndrs/jotai#2399] [Bug report]
## Summary
Jotai suspends whenever a derived atom reads a promise based atom even if that promise is already fulfilled. 

## Link to reproduction

https://codesandbox.io/p/sandbox/jotai-app-demo-with-v2-api-forked-mv5qxq

Clicking increment you'll see the flash of blue sometimes (or if you put a logpoint in jotai's `use` impl you can see the suspending directly )

Notice that consuming the resolved promise atom (_countAtom) directly doesn't suspend.

The reason this matters to me (if you need a real world use case) is that I'm using an async derived atom for the state of a MUI component and Suspense prevents the state from updating the component but doesn't prevent it's blur callback from firing resulting in the component reseting the state i just set on click back to the previous state.

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
> Jotai suspends whenever a derived atom reads a promise based atom even if that promise is already fulfilled.

It's an intentional design decision in Jotai v2.
It might not feel ideal, but basically we have three kinds of workarounds: https://github.com/pmndrs/jotai/discussions/2003#discussioncomment-6277700

**Reply by scamden** [reply]
Ah ok sorry to report as bug then. I'll take a look at the workarounds.
